net/veth.go

```diff
@@ -186,6 +186,15 @@ func AttachContainer(netNSPath, id, ifName, bridgeName string, mtu int, withMult
 		return err
 	}
 
+	// set hairpin mode
+	veth, err := netlink.LinkByName(name)
```

Comment:
ok, my mistake; the host end of the veth is returned from `CreateAndAttachVeth`

It's a callback so we can do initialization on the peer side before bringing the host side up.